src/EventListener/SwitchUserListener.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCacheBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCacheBundle\EventListener;
+
+use FOS\HttpCacheBundle\UserContextInvalidator;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Http\Event\SwitchUserEvent;
+use Symfony\Component\Security\Http\SecurityEvents;
+
+class SwitchUserListener implements EventSubscriberInterface
```

Comment:
can you add a phpdoc with the explanation why we do this? (basically the TLDR of our discussion in the issue you reported)